mlflow/gateway/providers/bedrock.py

```diff
@@ -0,0 +1,279 @@
+                "prompt": prompt,
+                **rename_payload_keys(
+                    payload,
+                    {
+                        "stop": "stop_sequences",
+                        "candidate_count": "num_generations",
+                    },
+                ),
+            }
+            # The range of Cohere's temperature is 0-5, but ours is 0-1, so we scale it.
+            if "temperature" in preped:
+                preped["temperature"] *= 5
+        else:
+            raise HTTPException(
+                status_code=422,
+                detail=f"Unknown AWS Bedrock model type {self._underlying_provider}",
+            )
+
+        return preped
+
+    def _process_output(self, response: dict):
+        if self._underlying_provider is AWSBedrockModelProvider.ANTHROPIC:
+            return {
+                "candidates": [
+                    {
+                        "text": response.get("completion"),
+                        "metadata": {
+                            "finish_reason": "stop"
+                            if response.get("stop_reason") == "stop_sequence"
+                            else "length"
+                        },
+                    }
+                ]
+            }
+        elif self._underlying_provider is AWSBedrockModelProvider.AMAZON:
+            return {
+                "candidates": [
+                    {"text": candidate.get("outputText"), "metadata": {}}
+                    for candidate in response.get("results", [])
+                ]
+            }
+        elif self._underlying_provider is AWSBedrockModelProvider.AI21:
+            return {
+                "candidates": [
+                    # second .get ensures item only has key "text",
+                    # but might be redundant/undesirable
+                    {"text": candidate.get("data", {}).get("text"), "metadata": {}}
+                    for candidate in response.get("completions", [])
+                ]
+            }
+        elif self._underlying_provider is AWSBedrockModelProvider.COHERE:
+            return {
+                "candidates": [
+                    {"text": candidate.get("text"), "metadata": {}}
+                    for candidate in response.get("generations", [])
+                ]
+            }
+
+        else:
+            raise HTTPException(
+                status_code=422,
+                detail=f"Unknown AWS Bedrock model type {self._underlying_provider}",
+            )
+
+    def _make_request(self, body):
+        try:
+            response = self.get_bedrock_client().invoke_model(
+                body=json.dumps(body).encode(),
+                modelId=self.config.model.name,
+                # defaults
+                # save=False,
+                accept="application/json",
+                contentType="application/json",
+            )
+            return json.loads(response.get("body").read())
+        # except botocore.exceptions.ValidationException as e:
```

Comment:
Remove, if not needed

Reply:
I would like to catch it to raise as a 422, but botocore exceptions are a PITA to catch.  I'll add a TODO